bin/wasp

```diff
@@ -34,36 +32,26 @@ do {
 	exit(1);
 } while (0);
 
+// create some services
 $input = new ArgvInput();
 $output = new ConsoleOutput();
 $errOut = $output->getErrorOutput();
 // default to decoration for the logger (can be disabled with --no-ansi)
 $errOut->setDecorated(true);
 $logger = new ConsoleLogger($errOut);
-$application = App::create($input, $output, $logger);
 
 // look for any plugins inside of the lock file
 $lockFile = getenv('WASP_LOCK') ?: $lockFile;
 // do we have one?
 if (file_exists($lockFile)) {
-	App::setPlugins($application, file_get_contents($lockFile));
+	$lock = file_get_contents($lockFile);
 } else {
-	$application->services->logger->warning('No lock file located');
+	$logger->warning('No lock file located');
+	$lock = null;
 }
 
-// add global options
-$application->getDefinition()->addOption(
-	new InputOption('include', null, InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'Additional files to include before executing.')
-);
-// process global options before running command
-$application->services->dispatcher->addListener(ConsoleEvents::COMMAND, function ($event) use ($application) {
-	// optional files to include
-	foreach ($application->services->input->getOption('include') as $file) {
-		if (!is_readable($file)) {
-			throw new RuntimeException("File does not exist or is not readable: $file");
-		}
-		require_once $file;
-	}
-});
+// set up the application
+$wasp = new Wasp(compact('input', 'output', 'logger'), $lock);
```

Comment:
Why not use the application defaults for these? Especially the logger since it seems like ConsoleLogger is already the default

Reply:
Input _could_ use a default such as `ArgvInput`, but I decided against it so that the application has to explicitly decide what kind of input to use (the demo is going to use `StringInput`, so that default would only be useful here anyway). As for `ConsoleLogger`, we need to instantiate it ourselves so we can pass an `OutputInterface` that writes to StdErr for all log messages, and as a result, we must instantiate `ConsoleOutput` to grab the error output object and turn on decoration by default

Reply:
(see L#37-41 to see what I mean)

Reply:
Oh yep, totally missed that!